test/unit/api/layer.test.js

```diff
@@ -97,30 +97,33 @@ describe('api/layer', () => {
     });
 
     describe('.blendToViz', () => {
-        it('should not throw an error if a valid viz is passed', () => {
+        it('should resolve the promise if a valid viz is passed', (done) => {
             const layer = new Layer('layer0', source, viz);
-            expect(function () {
-                layer.blendToViz(viz2);
-            }).not.toThrow();
+            // Mock for _vizChanged
```

Comment:
This is testing the implementation

Reply:
This is actually testing that `_checkId` passes without errors. So `_vizChanged` should be mocked.